File: test_regress/t/t_case_huge.pl
#!/usr/bin/perl
if (!$::Driver) { use FindBin; exec("./driver.pl", @ARGV, $0); die; }
# $Id$
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2003 by Wilson Snyder. This program is free software; you can
# redistribute it and/or modify it under the terms of either the GNU
# General Public License or the Perl Artistic License.

compile (
	 v_flags2 => ["--stats --profile-cfuncs"],
	 );

if ($Last_Self->{v3}) {
    file_grep ($Last_Self->{stats}, qr/Optimizations, Tables created\s+10/i);
    file_grep ($Last_Self->{stats}, qr/Optimizations, Combined CFuncs\s+10/i);
}

execute (
	 check_finished=>1,
     );

ok(1);
1;

File: test_regress/t/t_case_huge.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# This program is free software; you can redistribute it and/or modify it
# under the terms of either the GNU Lesser General Public License Version 3
# or the Perl Artistic License Version 2.0.
# SPDX-FileCopyrightText: 2024 Wilson Snyder
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('simulator')

# This tests combining CFuncs, but Dfg would inline the submodule, disabling
test.compile(verilator_flags2=["--stats", "-fno-dfg"])

test.execute()

test.file_grep(test.stats, r'Optimizations, Cases table normal\s+(\d+)', 8)
test.file_grep(test.stats, r'Optimizations, Cases table tiny\s+(\d+)', 0)
test.file_grep(test.stats, r'Optimizations, Cases parallelized\s+(\d+)', 3)
test.file_grep(test.stats, r'Optimizations, Tables created\s+(\d+)', 2)
test.file_grep(test.stats, r'Optimizations, Combined CFuncs\s+(\d+)', 9 if test.vltmt else 8)

test.passes()
